tests/test_modeling_tf_common.py

```diff
@@ -380,65 +366,191 @@ def test_pt_tf_model_equivalence(self):
+            return pt_inputs_dict
+
+        def check_outputs(tfo, pto, model_class, names):
+            """
+            Args:
+                model_class: The class of the model that is currently testing. For example, `TFBertModel`,
+                    TFBertForMaskedLM`, `TFBertForSequenceClassification`, etc. Currently unused, but it could make
+                    debugging easier and faster.
+
+                names: A string, or a list of strings. These specify what tfo/pto represent in the model outputs.
+                    Currently unused, but in the future, we could use this information to make the error message clearer
+                    by giving the name(s) of the output tensor(s) with large difference(s) between PT and TF.
+            """
+
+            # Some issue (`about past_key_values`) to solve (e.g. `TFPegasusForConditionalGeneration`) in a separate PR.
+            if names == "past_key_values":
+                return
+
+            if type(tfo) == tuple:
+                self.assertEqual(type(pto), tuple)
+                self.assertEqual(len(tfo), len(pto))
+                if type(names) in [tuple, list]:
+                    for to, po, name in zip(tfo, pto, names):
+                        check_outputs(to, po, model_class, names=name)
+                elif type(names) == str:
+                    for idx, (to, po) in enumerate(zip(tfo, pto)):
+                        check_outputs(to, po, model_class, names=f"{names}_{idx}")
+            elif isinstance(tfo, tf.Tensor):
+                self.assertTrue(isinstance(pto, torch.Tensor))
+
+                tfo = tfo.numpy()
+                pto = pto.numpy()
+
+                tf_nans = np.copy(np.isnan(tfo))
+                pt_nans = np.copy(np.isnan(pto))
+
+                pto[tf_nans] = 0
+                tfo[tf_nans] = 0
+                pto[pt_nans] = 0
+                tfo[pt_nans] = 0
+
+                max_diff = np.amax(np.abs(tfo - pto))
+                self.assertLessEqual(max_diff, 1e-5)
 
-            tf_hidden_states = tfo[0].numpy()
-            pt_hidden_states = pto[0].numpy()
+        for model_class in self.all_model_classes:
 
-            tf_nans = np.copy(np.isnan(tf_hidden_states))
-            pt_nans = np.copy(np.isnan(pt_hidden_states))
+            config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
 
-            pt_hidden_states[tf_nans] = 0
-            tf_hidden_states[tf_nans] = 0
-            pt_hidden_states[pt_nans] = 0
-            tf_hidden_states[pt_nans] = 0
+            # Output all for aggressive testing
+            config.output_hidden_states = True
+            # Pure convolutional models have no attention
+            # TODO: use a better and general criteria
+            if "TFConvNext" not in model_class.__name__:
+                config.output_attentions = True
+
+            # TODO: remove this block once the large negative value for attention masks is fixed.
```

Comment:
Would you like to take a stab at that? :) (If yes, it should be in a separate PR, of course)

Reply:
### 

> Would you like to take a stab at that? :) (If yes, it should be in a separate PR, of course)

Yes, it is on my plans :)